network-config-analyzer/IstioSidecarYamlParser.py

```diff
@@ -0,0 +1,233 @@
+
+        # currently dnsName pattern supported is of "internal" services only (i.e. k8s service)
+        # the dnsName in this case form looks like <service_name>.<domain>.svc.cluster.local
+        # also FQDN is of the format [hostname].[domain].[tld]
+        fqdn_regex = "^((?!-)[A-Za-z0-9-]+(?<!-)\\.)+[A-Za-z]+"
+        if alphabet_str and not re.fullmatch(fqdn_regex, alphabet_str):
+            self.syntax_error(f'Illegal host value pattern: {dns_name}, '
+                              f'dnsName should be specified using FQDN format', self)
+
+    def _get_peers_from_host_dns_name(self, dns_name):
+        """
+        Return the workload instances of the service in the given dns_name
+        :param str dns_name: the service given in the host
+        :rtype: PeerSet
+        """
+        # supported dns_name format is fqdn with internal k8s services pattern
+        # <service_name>.<domain>.svc.cluster.local
+        # the only relevant part of it is <service_name>
+        service_name = dns_name.split('.')[0]
+        return self.peer_container.get_pods_with_service_name_containing_given_string(service_name)
+
+    def _parse_egress_rule(self, egress_rule):
+        """
+        Parse a single egress rule, producing a IstioSidecarRule
+        :param dict egress_rule: The dict with the egress rule (IstioEgressListener) fields
+        :return: A IstioSidecarRule with the proper PeerSet
+        :rtype: IstioSidecarRule
+        """
+        allowed_keys = {'port': [0, dict], 'bind': [0, str], 'captureMode': [0, str], 'hosts': [1, list]}
+        self.check_fields_validity(egress_rule, 'Istio Egress Listener', allowed_keys)
+        # currently only hosts is considered in the rule parsing
+        if egress_rule.get('port'):
+            self._parse_port(egress_rule.get('port'))  # calling this for syntax checking only
+
+        hosts = egress_rule.get('hosts')
+        if not hosts:
+            self.syntax_error('One or more service hosts to be exposed by the listener are required', self)
+        res_peers = PeerSet()
+        for host in hosts:
+            host_peers = PeerSet()
+            # Services in the specified namespace matching dnsName will be exposed.
+            namespace, dns_name = self._validate_and_partition_host_format(host)
+            host_peers |= self._get_peers_from_host_namespace(namespace)
+            self._validate_dns_name_pattern(dns_name)
+            if host_peers:  # if there are services in the specified namespace
+                if '*' not in dns_name:  # * means all services in namespace, all already in host_peers
+                    host_peers &= self._get_peers_from_host_dns_name(dns_name)
+            res_peers |= host_peers
+
+        return IstioSidecarRule(res_peers)
+
+    def parse_policy(self):
+        """
+        Parses the input object to create a IstioSidecar object
+        :return: a IstioSidecar object with proper PeerSets
+        :rtype: IstioSidecar
+        """
```

Comment:
done

Reply:
Hi @shireenf-ibm , as we discussed, we had preferred to avoid handling this in the parsing stage,  creating sidecar policy objects that are fully equivalent to the input resources, and considering this "override" behavior in the stage of the allowed connections computation..
Is it possible to consider this other alternative? (e.g. sort the sidcar policy objects by priority, to consider only the most specific that is relevant?)

Reply:
oops, I might missed that point.
I considered the solution with the less memory and runtime consuming, (since "first" injected sidecar (with selector) selecting a workload is always the only one considered for that workload)
However, In this branch I have also added many tests (for as much cases - on bookinfo) and confirmed all of  tests behaviors on the live cluster.
So , I'll open a new branch from this one and try to implement a solution on the stage of the allowed connections computation..

Reply:
done - closing this PR, and opening a new one with calculating priority in allowed_connections stage - the new PR's branch is derived from this